packages/@ourworldindata/grapher/src/lineLegend/LineLegend.tsx

```diff
@@ -360,49 +372,163 @@ export class LineLegend extends React.Component<{
+                this.intialSeriesByName.get(seriesName)
+            )
+        )
     }
 
-    @computed get placedSeries(): PlacedSeries[] {
+    @computed get partialInitialSeries(): PlacedSeries[] {
+        const availableHeight = this.manager.yAxis.rangeSize
         const nonOverlappingMinHeight =
             sumBy(this.initialSeries, (series) => series.bounds.height) +
             this.initialSeries.length * LEGEND_ITEM_MIN_SPACING
-        const availableHeight = this.manager.yAxis.rangeSize
-        if (nonOverlappingMinHeight > availableHeight)
-            return this.overlappingPlacement
-        return this.standardPlacement
+
+        // early return if filtering is not needed
+        if (nonOverlappingMinHeight <= availableHeight)
+            return this.initialSeries
+
+        if (this.sortedSeriesByImportance) {
+            // keep a subset of series that fit within the available height,
+            // prioritizing by importance. Note that more important (but longer)
+            // series names are skipped if they don't fit.
+            const keepSeries: PlacedSeries[] = []
+            let keepSeriesHeight = 0
+            for (const series of this.sortedSeriesByImportance) {
+                const newHeight =
+                    keepSeriesHeight +
+                    series.bounds.height +
+                    LEGEND_ITEM_MIN_SPACING
+                if (newHeight <= availableHeight) {
+                    keepSeries.push(series)
+                    keepSeriesHeight = newHeight
+                    if (keepSeriesHeight > availableHeight) break
+                }
+            }
+            return keepSeries
+        } else {
+            const candidates = new Set<PlacedSeries>(this.initialSeries)
+            const sortedKeepSeries: PlacedSeries[] = []
+
+            let keepSeriesHeight = 0
+
+            const maybePickCandidate = (candidate: PlacedSeries): boolean => {
+                const newHeight =
+                    keepSeriesHeight +
+                    candidate.bounds.height +
+                    LEGEND_ITEM_MIN_SPACING
+                if (newHeight <= availableHeight) {
+                    const insertIndex = sortedIndexBy(
+                        sortedKeepSeries,
+                        candidate,
+                        (s) => s.midY
+                    )
+                    sortedKeepSeries.splice(insertIndex, 0, candidate)
+                    candidates.delete(candidate)
+                    keepSeriesHeight = newHeight
+                    return true
+                }
+                return false
+            }
+
+            type Bracket = [number, number]
+            const findBracket = (
+                sortedBrackets: Bracket[],
+                n: number
+            ): [number | undefined, number | undefined] => {
+                const firstBracketValue = sortedBrackets[0][0]
+                const lastBracketValue = last(sortedBrackets)![1]
```

Comment:
At unreasonably small sizes these lines can fail, because there are no brackets at all (and hence `sortedBrackets[0]` and `last(sortedBrackets)` are both `undefined`):

http://staging-site-line-legend-algo/grapher/share-in-poverty-relative-to-different-poverty-thresholds is failing for me at size 237 x 507.

Reply:
Good catch!